packages/frontend-shared/src/gql-components/error/BaseError.vue

```diff
@@ -148,6 +149,14 @@ const markdownTarget = ref()
 const baseError = computed(() => props.gql)
 const { markdown } = useMarkdown(markdownTarget, computed(() => props.gql.errorMessage), { classes: { code: ['bg-error-200'] } })
 
+const retrying = ref(false)
+
+const onRetry = async () => {
+  retrying.value = true
+  await props.retry?.(baseError.value.id)
```

Comment:
Change the type of props.retry to `(id: string) => Promise<void>`

Reply:
I just noticed we pass a function as a prop and handle it. This is unusual - normally you'd do `emit("retry")` then receive the result as a prop, instead of awaiting a no-op (assuming `Promise<void>` is actually the correct return type). 

Edit: this will not always be a promise: https://github.com/cypress-io/cypress/pull/22514/files#diff-b580358883f946c09563d80b0301fc0e521b5ff8435513cc5abb9c671f655dccR156

I know this wasn't introduced here, but we should probably change it to emit and event... but not a blocker.

Reply:
How about this:

1. define new emit, something like `(event: 'retry', id: string): void`
2. `emit("retry", baseError.value.id)` here
3. in the parent, listen for `@retry="resetErrorAndLoadConfig"
4. In there, do something like:

```js
const retrying = ref(false)
const resetErrorAndLoadConfig = async (id: string) => {
  retrying.value = true
  if (!mutation.fetching.value) {
    await mutation.executeMutation({ id }) // do we care about result of this? I don't think so
  }
  retrying.value = false
}
```

5. Pass `retrying` prop to `BaseError`.

We'd be lifting the state from BaseError into the parent (as they call it in the [React docs](https://reactjs.org/docs/lifting-state-up.html). I wonder if BaseError should be the one to decide if it's errored, or it should be a "dumb" or "presentation" component only?

Thoughts @astone123 @ZachJW34 ? Bit of work, but I think it's good to start attacking some of the debt we introduced while learning Vue and building the unification project.

Reply:
@lmiller1990 I'm definitely up for making this more idiomatic. With React, I'm used to passing functions down from parent to child to handle state changes like this, but I agree that it would be better to "lift state" into the parent (talking about the `retrying` state).

Will work on this

Reply:
The emit makes more sense here so I think it's a good idea and worth implementing. Can we remove the extra `retrying` ref and just use the value of `mutation.fetching.value` as our loading state?